platform/ext/target/nordic_nrf/common/nrf_common.cmake

```diff
@@ -5,12 +5,145 @@
 #
 #-------------------------------------------------------------------------------
 
-set(NRF_COMMON_DIR ${CMAKE_CURRENT_LIST_DIR})
+set(NRF_COMMON_DIR "${CMAKE_CURRENT_LIST_DIR}")
 set(NRFX_DIR "${NRF_COMMON_DIR}/nrfx")
 
+#Specify the location of platform specific build dependencies.
+if(COMPILER STREQUAL "ARMCLANG")
+    set (S_SCATTER_FILE_NAME   "${PLATFORM_DIR}/common/armclang/tfm_common_s.sct")
+    set (BL2_SCATTER_FILE_NAME "${NRF_COMMON_DIR}/armclang/nordic_nrf_bl2.sct")
+    set (NS_SCATTER_FILE_NAME  "${NRF_COMMON_DIR}/armclang/nordic_nrf_ns.sct")
+    if (DEFINED CMSIS_5_DIR)
+      # not all project defines CMSIS_5_DIR, only the ones that use it.
+      set (RTX_LIB_PATH "${CMSIS_5_DIR}/CMSIS/RTOS2/RTX/Library/ARM/RTX_V8MBN.lib")
+    endif()
+elseif(COMPILER STREQUAL "GNUARM")
+    set (S_SCATTER_FILE_NAME   "${PLATFORM_DIR}/common/gcc/tfm_common_s.ld")
+    set (BL2_SCATTER_FILE_NAME "${NRF_COMMON_DIR}/gcc/nordic_nrf_bl2.ld")
+    set (NS_SCATTER_FILE_NAME  "${NRF_COMMON_DIR}/gcc/nordic_nrf_ns.ld")
+    if (DEFINED CMSIS_5_DIR)
+      # not all project defines CMSIS_5_DIR, only the ones that use it.
+      set (RTX_LIB_PATH "${CMSIS_5_DIR}/CMSIS/RTOS2/RTX/Library/GCC/libRTX_V8MBN.a")
```

Comment:
...and here

Reply:
Alternatively, this could be moved in the specific SoC .cmake file (if we think Nordic will eventually have a cortex-m23) :)

Reply:
Hehe :), if you do prefer it, I will move it 👍.

Reply:
Not really - i find it unlikely we will need to account for Baseline v8-m architecture
But we need the fix i mention.